### PR TITLE
feat(spinner): add determinate spinner

### PR DESCRIPTION
**Pull request by sandgraham**
Adds `SpinnerDeterminate` to the `spinner` module.

Kapture 2020-10-22 at 12 41 29

TODO:
- [x] Basic implementation
- [x] Animation
- [x] Sizes
- [x] Add overrides
- [x] Add documentation

### CODE REVIEW

**Review by chasestarr — APPROVED** [review]
works well!

File: documentation-site/examples/spinner/determinate.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+// @flow
+import * as React from 'react';
+import {Button} from 'baseui/button';
+import {SpinnerDeterminate} from 'baseui/spinner';
+import {useStyletron} from 'baseui';
+
+export default () => {
+  const [css] = useStyletron();
+  // Mimic some loading
+  const [x, xx] = React.useState(0);
```
Comment (tajo) [inline documentation-site/examples/spinner/determinate.js:10]: `x`, `xx` and `foo`, what am I seeing here haha?

**Comment by tajo** [comment]
Why is this spinner and not a progress bar? Should we add `role="progressbar"` and related aria attrs?

File: documentation-site/examples/spinner/determinate.js (hunk 1 above)
Comment (sandgraham) [reply documentation-site/examples/spinner/determinate.js:10]: whoops - forgot to rename those 😅

**Comment by sandgraham** [comment]
> Why is this spinner and not a progress bar? Should we add `role="progressbar"` and related aria attrs?

Added aria props but will keep it in spinner module since design has them documented together.

**Pull request merged**